File: 4th Semester/Systems for Design and Implementation/C#/Homework2/OnwardTravel/Models/Ride.cs
namespace OnwardTravel.Models
{
    internal class Ride : Entity<Guid>
    {
        public string Destination { get; }
        public DateTime Departure { get; }

        public Ride(Guid guid, string destination, DateTime departure)
            : base(guid)
        {
            Destination = destination;
            Departure = departure;
        }

        public Ride(string destination, DateTime departure)
        {
            Destination = destination;
            Departure = departure;
        }
    }
}

File: 4th Semester/Systems for Design and Implementation/Java/Homework4/OnwardModels/src/main/java/models/Ride.java
package models;

import java.time.LocalDateTime;
import java.util.UUID;

public class Ride extends Entity<UUID> {
    private final String destination;
    private final LocalDateTime departure;

    public Ride(UUID uuid, String destination, LocalDateTime departure) {
        super(uuid);
        this.destination = destination;
        this.departure = departure;
    }

    public Ride(String destination, LocalDateTime departure) {
        this.destination = destination;
        this.departure = departure;
    }

    public String getDestination() {
        return destination;
    }

    public LocalDateTime getDeparture() {
        return departure;
    }
}
